### ISSUE TITLE
With statements report their width as their length

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [Microsoft/TypeScript#20524]
**TypeScript Version**: 2.6.1

Found while writing a TSLint rule...

Expected failures:
```typescript
const foo = {
    a: "b"
};

with (foo) {
~~~~~~~~~~~~  [`with` statements are disallowed]
    // ...
}

const callback = () => {
    with (callback) {
    ~~~~~~~~~~~~~~~~~  [`with` statements are disallowed]
        // ...
    }
};
```

Linter:
```typescript
function walk(context: Lint.WalkContext<void>) {
    const callback = (node: ts.Node): void => {
        if (ts.isWithStatement(node)) {
            context.addFailureAt(node.getStart(), node.getWidth(), Rule.FAILURE_STRING);
        }

        return ts.forEachChild(node, callback);
    };

    return ts.forEachChild(context.sourceFile, callback);
}
```

**Expected:**
Statement widths should be reported as their first line width.

**Actual:**
`node.getWidth() === node.getText().length`

**Comment by JoshuaKGoldberg** [comment]
Oh never mind, that's the expected behavior :smile:
